week3/main.cpp

```diff
@@ -0,0 +1,139 @@
+//CodeU: Assignment_3
+//
+//  Created by Julia Semavina on 8.06.2017
+//
+//  problem specification:
+//      Given a grid of letters and a dictionary, find all the words from the dictionary that can be
+//      formed in the grid.
+//
+//  idea:
+//      We start building words from every cell of the grid. So, at first we have words of the length=1. If the word
+//      is prefix, we add one letter (from the North, South, East, West, N-W, N-E, S-W, S-E) if it is possible. Now we
+//      have words of the length=2.
+//      But this way if the input grid consists of the same items and input dictionary contains a very long string
```

Comment:
It's "a bit" more complex than that :)
C is the average number of branches, that starts at 8, 6 or 4 depending if it's center / lateral / corner, and goes down to 0 as more of the board is explored...
A google search reveals that there are 12 million possible words on a 4x4 grid, so that puts C in around 2.33 in that case :)